e3/text.py

```diff
@@ -20,3 +20,26 @@ def format_with_dict(pattern, values):
     key_regexp = r"|".join([r'\(%s\)' % k for k in values])
     result = re.sub(r'%%(?!%s)' % key_regexp, r'%%', pattern)
     return result % values
+
+
+def bytes_as_str(content):
+    """Safely convert bytes to a python string.
+
+    The conversion function attempt to convert to decode utf-8 and
```

Comment:
A bit hard to process, at least for the first line... How about?

This function attempts the conversion assuming a utf-8 enconding.
If this triggers a conversion error, it then falls back to a safe representation,
where some characters might be escaped.